### DISCUSSION TITLE
Is there a way to implement readonly files in Theia?

### DISCUSSION THREAD

**Discussion by vinayb21** [eclipse-theia/theia#9423] [Q&A]
Mark files as readonly and when the file tree is loaded, the files should be shown with a readonly file icon (or) a label should be shown that this file is a readonly file.

**Comment by akosyakov** [comment]
Only with a custom extension. You will need to customize `NavigatorWidget` to render such files differently. You also will need to customize `FileResourceProvider` to avoid creation of `FileResource` for such files and instead provide a custom `ResourceProvider` creating readonly `Resource`. See https://github.com/theia-ide/theia/blob/master/packages/core/src/common/resource.ts
